compiler/sem/semstmts.nim

```diff
@@ -1099,155 +1074,425 @@ proc semLetOrVar(c: PContext, n: PNode, symkind: TSymKind): PNode =
+    # set the symbol type and add the symbol to the production
+    producedDecl[i] =
+      if v.typ != nil and not sameTypeOrNil(v.typ, vTyp):
+
+        c.config.newError(
+          r,
+          SemReport(
+            kind: rsemDifferentTypeForReintroducedSymbol,
+            sym: v,
+            typeMismatch: @[c.config.typeMismatch(
+              actual = vTyp, formal = v.typ)]))
+      else:
+        v.typ = vTyp
+
+        newSymNode2(v)
+
+    case def.kind
+    of nkEmpty:
+      let actualType = v.typ.skipTypes({tyGenericInst, tyAlias,
+                                        tyUserTypeClassInst})
+
+      producedDecl[i] =
+        if actualType.kind in {tyObject, tyDistinct} and
+            actualType.requiresInit:
+          defaultConstructionError2(c, v.typ, r)
+        else:
+          checkNilableOrError(c, producedDecl[i])
+    else:
+      if sfThread in v.flags:
+        hasError = true
+        producedDecl[^1] = c.config.newError(n[^1],
+                                             reportSem rsemThreadvarCannotInit)
+    
+    if producedDecl[i].isError:
+      hasError = true
+
+    if not hasError and sfCompileTime in v.flags:
+      var x = newNodeI(result.kind, v.info)
+      x.add producedDecl
+      vm.setupCompileTimeVar(c.module, c.idgen, c.graph, x)
+
+    if v.flags * {sfGlobal, sfThread} == {sfGlobal}:
+      # this is just logging, doesn't need to be converted to an error
+      localReport(c.config, v.info, reportSym(rsemGlobalVar, v))
+
+  case defPart.kind
+  of nkVarTuple:
+    # these transform `b` into an `nkError` if required, we use this when
+    # populating all the children
+    result.add:
+      if tupTyp.kind != tyTuple:
+        hasError = true
+        newError(
+              c.config,
+              producedDecl,
+              c.config.semReportTypeMismatch(producedDecl, {tyTuple}, tupTyp))
+      elif not sameTupleTypeAndDefAirity:
+        hasError = true
+        newError(
+              c.config,
+              producedDecl,
+              semReportCountMismatch(
+                rsemWrongNumberOfVariables,
+                expected = defCount,
+                got = tupTyp.len, node = producedDecl))
+      else:
+        producedDecl
+
+  of nkConstDef:
+    # we rely on the fact that the const definitions are singletons
+    
+    result.add producedDecl
+
+    if tupTyp.kind == tyTuple and def.kind in {nkPar, nkTupleConstr} and
+        defPart.len > 3:
+      # xxx: helpful hints like this should likely be associated to an id/pos
+      #      instead of reporting
+      localReport(c.config, defPart.info, reportSem rsemEachIdentIsTuple)
+  else:
+    discard
 
   if hasError:
-    # xxx: hasError is here until we've converted the whole proc to use nkError
-    #      until then we check at the end to ensure we capture any nkErrors
-    #      embedded within the section
+    # wrap the result if there is an embedded error
     result = c.config.wrapError(result)
-
+    
```

Comment:
```suggestion

```

Reply:
I left an extra space between procs to more clearly demarcate the transition when quickly skimming, not married to it but curious what you think?

Reply:
I meant the trailing whitespace (I should have mentioned that with the suggestion), the extra line is fine and I agree that it helps with readability.